include/inexor/vulkan-renderer/exceptions/vk_exception.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "inexor/vulkan-renderer/exceptions/exception.hpp"
+
+#include <vulkan/vulkan_core.h>
+
+namespace inexor::vulkan_renderer::exceptions {
+
+/// @brief
+class VulkanException final : public Exception {
+
+    ///
+    [[nodiscard]] std::string get_vkresult_description(const VkResult result) const;
+
+    ///
+    [[nodiscard]] std::string get_vkresult_string(const VkResult result) const;
+
+public:
+    /// @brief
+    /// @param result
+    /// @param message
+    VulkanException(const VkResult result, const std::string message);
```

Comment:
I would switch the parameters.
```suggestion
    VulkanException(const std::string message, const VkResult result);
```

Reply:
Done.